adding ability to pass env variable to kaniko pods

prary opened the pull request:
Fixes #3227  

**Description**
adding ability to pass env variable to kaniko pods

**User facing changes**
```
cluster:
  envs:
  - name: JAVA_PATH
     value: /usr/bin
  - name: JAVA_PATH
     value: /usr/bin
```

**Next PRs.**

n/a

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Mentions any output changes.
- [ ** ] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [ ] Adds integration tests if needed.


**Reviewer Notes**

- [ ** ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.


**Release Notes**

<!-- 
Now user can add envrionment to kaniko pod through skaffold.yaml
-->

balopat commented on pkg/skaffold/schema/latest/config.go:
```diff
@@ -310,6 +312,9 @@ type ClusterDetails struct {
 	// Defaults to 0.
 	Concurrency int `yaml:"concurrency,omitempty"`
+
+	// Envs is array of all the environment variable to be passed to kaniko.
```
```suggestion
	// Envs is array of all the environment variable to be passed to the pod building _all_ of the artifacts.
```

prary commented:
@balopat thanks for pointing out 
> Also if we add this at the build.cluster level, it will apply to all artifact

Updated the PR and added test case for the same.

balopat commented on docs/content/en/docs/pipeline-stages/builders/docker.md:
````diff
@@ -92,6 +92,16 @@ build:
 ```
 Note that the Kubernetes secret must not be of type `kubernetes.io/dockerconfigjson` which stores the config json under the key `".dockerconfigjson"`, but an opaque secret with the key `"config.json"`.
 
+To set up envrionment variable for Kaniko pod.
````
```suggestion
You can set up environment variables for each artifact's Kaniko pod:
```

balopat commented:
I'm sorry @prary for making you work on this - In the meantime @cyrildiagne opened this up in https://github.com/GoogleContainerTools/skaffold/pull/3389 and I forgot that I had yours open (apologies for my absent-mindedness) - @cyrildiagne's PR was also a rebase of the older PR https://github.com/GoogleContainerTools/skaffold/pull/3115. 
If you agree, I'd rebase this PR to just contain the docs part that is still useful!

prary commented:
Hi @balopat no worries go ahead as you feel the right way :)

prary commented:
Should I update the pr with just documentation only?

dgageot commented:
Hi @prary  I'm afraid kaniko code has changed a lot since then. Could you try to rebase your PR? If you can't, let me know, I might be able to continue your work

dgageot commented:
Closing as a duplicate of #3287
